File: Actividad 1/4_1.py
class Tienda:
    def __init__(self, nombre, direccion, tipo):
        self.nombre = nombre
        self.direccion = direccion
        self.tipo = tipo

    def vender_productos(self):
        producto = input("¿Qué producto desea vender? ")
        cantidad = int(input("Cantidad de productos: "))
        print("¡Producto vendido!")

    def agregar_productos(self):
        producto = input("¿Qué producto desea agregar? ")
        cantidad = int(input("Cantidad de productos: "))
        print("¡Producto agregado!")


class Estudiante:
    def __init__(self, nombres, apellidos, colegio, edad):
        self.nombres = nombres
        self.apellidos = apellidos
        self.colegio = colegio
        self.edad = edad


def menu_principal():
    print("Menu")
    print("1. Generar estudiante")
    print("2. Manejar estudiante")
    print("3. Salir")
    return int(input("Elija una opción: "))


def menu_estudiante():
    print("Menu")
    print("1. Crear tienda")
    print("2. Agregar productos")
    print("3. Vender productos")
    print("4. Salir")
    return int(input("Elija una opción: "))


tiendas = []
estudiantes = []

while True:
    opcion = menu_principal()
    if opcion == 1:
        nombres = input("Nombres del estudiante: ")
        apellidos = input("Apellidos del estudiante: ")
        colegio = input("Colegio del estudiante: ")
        edad = int(input("Edad del estudiante: "))
        estudiante = Estudiante(nombres, apellidos, colegio, edad)
        estudiantes.append(estudiante)
    elif opcion == 2:
        nombres = input("Ingrese el nombre del estudiante: ")
        for estudiante in estudiantes:
            if estudiante.nombres == nombres:
                while True:
                    opcion = menu_estudiante()
                    if opcion == 1:
                        nombre = input("Nombre de la tienda: ")
                        direccion = input("Dirección de la tienda: ")
                        tipo = input("Tipo de tienda: ")
                        tienda = Tienda(nombre, direccion, tipo)
                        tiendas.append(tienda)
                        break
                    elif opcion == 2:
                        nombre = input("Ingrese el nombre de la tienda: ")
                        for tienda in tiendas:
                            if tienda.nombre == nombre:
                                tienda.agregar_productos()
                                break
                        else:
                            print("Tienda no encontrada")
                    elif opcion == 3:
                        nombre = input("Ingrese el nombre de la tienda: ")
                        for tienda in tiendas:
                            if tienda.nombre == nombre:
                                tienda.vender_productos()
                                break
                        else:
                            print("Tienda no encontrada")
                    elif opcion == 4:
                        print("Saliendo...")
                        break
    elif opcion == 3:
        print("Saliendo...")
        break

File: Actividad 1/4_1.java
import java.util.ArrayList;
import java.util.Scanner;

class Tienda {
    String nombre;
    String direccion;
    String tipo;

    public Tienda(String nombre, String direccion, String tipo) {
        this.nombre = nombre;
        this.direccion = direccion;
        this.tipo = tipo;
    }

    void venderProductos() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("¿Qué producto desea vender? ");
        String producto = scanner.nextLine();
        System.out.print("Cantidad de productos: ");
        int cantidad = scanner.nextInt();
        System.out.println("¡Producto vendido!");
    }

    void agregarProductos() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("¿Qué producto desea agregar? ");
        String producto = scanner.nextLine();
        System.out.print("Cantidad de productos: ");
        int cantidad = scanner.nextInt();
        System.out.println("¡Producto agregado!");
    }
}

class Estudiante {
    String nombres;
    String apellidos;
    String colegio;
    int edad;

    public Estudiante(String nombres, String apellidos, String colegio, int edad) {
        this.nombres = nombres;
        this.apellidos = apellidos;
        this.colegio = colegio;
        this.edad = edad;
    }
}

public class Main{
    static Scanner scanner = new Scanner(System.in);
    static ArrayList<Tienda> tiendas = new ArrayList<>();
    static ArrayList<Estudiante> estudiantes = new ArrayList<>();

    public static void main(String[] args) {
        while (true) {
            int opcion = menuPrincipal();
            if (opcion == 1) {
                System.out.print("Nombres del estudiante: ");
                String nombres = scanner.nextLine();
                System.out.print("Apellidos del estudiante: ");
                String apellidos = scanner.nextLine();
                System.out.print("Colegio del estudiante: ");
                String colegio = scanner.nextLine();
                System.out.print("Edad del estudiante: ");
                int edad = scanner.nextInt();
                scanner.nextLine(); // Consumir la nueva línea
                Estudiante estudiante = new Estudiante(nombres, apellidos, colegio, edad);
                estudiantes.add(estudiante);
            } else if (opcion == 2) {
                System.out.print("Ingrese el nombre del estudiante: ");
                String nombres = scanner.nextLine();
                for (Estudiante estudiante : estudiantes) {
                    if (estudiante.nombres.equals(nombres)) {
                        while (true) {
                            int opcionEstudiante = menuEstudiante();
                            if (opcionEstudiante == 1) {
                                System.out.print("Nombre de la tienda: ");
                                String nombre = scanner.nextLine();
                                System.out.print("Dirección de la tienda: ");
                                String direccion = scanner.nextLine();
                                System.out.print("Tipo de tienda: ");
                                String tipo = scanner.nextLine();
                                Tienda tienda = new Tienda(nombre, direccion, tipo);
                                tiendas.add(tienda);
                            } else if (opcionEstudiante == 2) {
                                System.out.print("Ingrese el nombre de la tienda: ");
                                String nombre = scanner.nextLine();
                                boolean encontrada = false;
                                for (Tienda tienda : tiendas) {
                                    if (tienda.nombre.equals(nombre)) {
                                        tienda.agregarProductos();
                                        encontrada = true;
                                        break;
                                    }
                                }
                                if (!encontrada) {
                                    System.out.println("Tienda no encontrada");
                                }
                            } else if (opcionEstudiante == 3) {
                                System.out.print("Ingrese el nombre de la tienda: ");
                                String nombre = scanner.nextLine();
                                boolean encontrada = false;
                                for (Tienda tienda : tiendas) {
                                    if (tienda.nombre.equals(nombre)) {
                                        tienda.venderProductos();
                                        encontrada = true;
                                        break;
                                    }
                                }
                                if (!encontrada) {
                                    System.out.println("Tienda no encontrada");
                                }
                            } else if (opcionEstudiante == 4) {
                                System.out.println("Saliendo...");
                                break;
                            }
                        }
                    } else {
                        System.out.println("Estudiante no encontrado");
                    }
                }
            } else if (opcion == 3) {
                System.out.println("Saliendo...");
                break;
            }
        }
    }

    static int menuPrincipal() {
        System.out.println("Menu");
        System.out.println("1. Generar estudiante");
        System.out.println("2. Manejar estudiante");
        System.out.println("3. Salir");
        System.out.print("Elija una opción: ");
        return Integer.parseInt(scanner.nextLine());
    }

    static int menuEstudiante() {
        System.out.println("Menu");
        System.out.println("1. Crear tienda");
        System.out.println("2. Agregar productos");
        System.out.println("3. Vender productos");
        System.out.println("4. Salir");
        System.out.print("Elija una opción: ");
        return Integer.parseInt(scanner.nextLine());
    }
}
